tests/regression/copy_stencil_tuple.cpp

```diff
@@ -0,0 +1,208 @@
+            eval(out3()) = eval(in3());
+        }
+    };
+
+    struct copy_functor_tuple {
+        using in = in_accessor<0>;
+        using out = inout_accessor<1>;
+
+        using param_list = make_param_list<in, out>;
+
+        template <class Eval>
+        GT_FUNCTION static void apply(Eval &&eval) {
+            eval(out()) = eval(in());
+        }
+    };
+
+    struct copy_functor_tuple_unrolled {
+        using in = in_accessor<0>;
+        using out = inout_accessor<1>;
+
+        using param_list = make_param_list<in, out>;
+
+        template <class Eval>
+        GT_FUNCTION static void apply(Eval &&eval) {
+            tuple_util::host_device::get<0>(eval(out())) = tuple_util::host_device::get<0>(eval(in()));
+            tuple_util::host_device::get<1>(eval(out())) = tuple_util::host_device::get<1>(eval(in()));
+            tuple_util::host_device::get<2>(eval(out())) = tuple_util::host_device::get<2>(eval(in()));
+            tuple_util::host_device::get<3>(eval(out())) = tuple_util::host_device::get<3>(eval(in()));
+        }
+    };
+
+    struct copy_functor_data_dims {
+        using in = in_accessor<0, extent<>, 4>;
+        using out = inout_accessor<1, extent<>, 4>;
+
+        using param_list = make_param_list<in, out>;
+
+        template <class Eval>
+        GT_FUNCTION static void apply(Eval &&eval) {
+            eval(out(0, 0, 0, 0)) = eval(in(0, 0, 0, 0));
+            eval(out(0, 0, 0, 1)) = eval(in(0, 0, 0, 1));
+            eval(out(0, 0, 0, 2)) = eval(in(0, 0, 0, 2));
+            eval(out(0, 0, 0, 3)) = eval(in(0, 0, 0, 3));
+        }
+    };
+
+    GT_REGRESSION_TEST(copy_stencil_tuple_unrolled, test_environment<>, stencil_backend_t) {
+        // baseline
+        auto in = [](int i, int j, int k) { return i + j + k; };
+        auto out0 = TypeParam::make_storage();
+        auto out1 = TypeParam::make_storage();
+        auto out2 = TypeParam::make_storage();
+        auto out3 = TypeParam::make_storage();
+        auto comp = [&out0,
+                        &out1,
+                        &out2,
+                        &out3,
+                        grid = TypeParam::make_grid(),
+                        in0 = TypeParam::make_const_storage(in),
+                        in1 = TypeParam::make_const_storage(in),
+                        in2 = TypeParam::make_const_storage(in),
+                        in3 = TypeParam::make_const_storage(in)] {
+            run_single_stage(
+                copy_functor_unrolled(), stencil_backend_t(), grid, in0, in1, in2, in3, out0, out1, out2, out3);
+        };
+        comp();
+        TypeParam::verify(in, out0);
+        TypeParam::verify(in, out1);
+        TypeParam::verify(in, out2);
+        TypeParam::verify(in, out3);
+        TypeParam::benchmark("copy_stencil_tuple_unrolled", comp);
+    }
+
+    GT_REGRESSION_TEST(copy_stencil_tuple_tuple, test_environment<>, stencil_backend_t) {
+        // expected to perform bad on GPU because array of structures
```

Comment:
As discussed: should only perform worse if the tuple is larger than 128bits or has a size which kills alignment (e.g., 5).